sdk/textanalytics/ai-text-analytics/README.md

```diff
@@ -36,64 +18,34 @@ Use the client library to:
-
-You can find the endpoint for your text analytics resource either in the [Azure Portal][azure_portal] or by using the [Azure CLI][azure_cli] snippet below:
-
-```bash
-az cognitiveservices account show --name <your-resource-name> --resource-group <your-resource-group-name> --query "endpoint"
-```
-
-#### Using an API Key
-
-Use the [Azure Portal][azure_portal] to browse to your Text Analytics resource and retrieve an API key, or use the [Azure CLI][azure_cli] snippet below:
+To create a client object to access the Text Analytics API, you will need the `endpoint` of your Text Analytics resource and a `credential`. The Azure Text Analytics Client can use Azure Active Directory credentials to authenticate.
 
-**Note:** Sometimes the API key is referred to as a "subscription key" or "subscription API key."
-
-```PowerShell
-az cognitiveservices account keys list --resource-group <your-resource-group-name> --name <your-resource-name>
-```
-
-Once you have an API key and endpoint, you can use the `AzureKeyCredential` class to authenticate the client as follows:
-
-```js
-const { TextAnalyticsClient, AzureKeyCredential } = require("@azure/ai-text-analytics");
-
-const client = new TextAnalyticsClient("<endpoint>", new AzureKeyCredential("<API key>"));
-```
+You can find the endpoint for your Text Analytics resource in the [Azure Portal][azure_portal].
 
 #### Using an Azure Active Directory Credential
 
-Client API key authentication is used in most of the examples, but you can also authenticate with Azure Active Directory using the [Azure Identity library][azure_identity]. To use the [DefaultAzureCredential][defaultazurecredential] provider shown below,
-or other credential providers provided with the Azure SDK, please install the `@azure/identity` package:
+Client API key authentication is used in most of the examples, but you can also authenticate with Azure Active Directory using the [Azure Identity library][azure_identity]. To use the [DefaultAzureCredential][defaultazurecredential] provider shown below, or other credential providers provided with the Azure SDK, please install the `@azure/identity` package:
 
 ```bash
 npm install @azure/identity
 ```
 
-You will also need to [register a new AAD application][register_aad_app] and grant access to Text Analytics by assigning the `"Cognitive Services User"` role to your service principal (note: other roles such as `"Owner"` will not grant the necessary permissions, only `"Cognitive Services User"` will suffice to run the examples and the sample code).
+You will also need to register a new AAD application and grant access to Text Analytics by assigning the suitable role to your service principal (note: roles such as `"Owner"` will not grant the necessary permissions).
```

Comment:
Do we need to do this if we were relying on the credentials that used Azure CLI or VS Code?

Reply:
I can remove this as well.

Reply:
@sadasant can you asnwer Ramya's question?

Reply:
Users don’t need to grant specific permission to the Azure CLI credential to use it to authenticate in the cases that I have tried. I have never tried Text Analytics though. I would make sure to corroborate first (by trying, let’s say, InteractiveBrowserCredential on NodeJS with no parameters)

Reply:
I'm taking a second look, one moment.

Reply:
So, DefaultAzureCredential tries: EnvironmentCredential, then ManagedIdentityCredential, then AzureCliCredential, then VisualStudioCodeCredential. From these:

- EnvironmentCredential doesn't have default values.
- ManagedIdentityCredential only has a default behavior if shipped on an environment with a system assigned credential (not local dev).
- AzureCliCredential assumes users authenticated using the Azure CLI.
- VisualStudioCodeCredential assumes users are trying this inside VSCode, with a specific extension installed and configured well.

This means that, the main reasonable scenario that doesn't need an AAD application is the Azure CLI case. For that case we would need to add a section asking users to authenticate using the CLI above this section.

The "no configuration" credentials are really InteractiveBrowser and DeviceCode, which are not yet in DefaultAzureCredential.

Reply:
Seems like in .Net, they always point users to create an AAD application: https://github.com/Azure/azure-sdk-for-net/tree/master/sdk/textanalytics/Azure.AI.TextAnalytics#create-textanalyticsclient-with-azure-active-directory-credential

I think this is a good default too. If we don't ask users to do `az login`, we should ask them to create an AAD application.

Reply:
From my understanding, you need to do the whole registering of application to get the "client id". So, all credentials that don't need client id should not need registering of the application either. That would be Azure CLI, VS Code and Managed Identity right?

Reply:
That is true. Other two that don't need this "client id" are InteractiveBrowser and DeviceCode. However, all of these credentials need some special setup: Azure CLI, VS Code and Managed Identity. If we don't provide instructions to create an AAD app, we should at least provide instructions to `az login` so that `DefaultAzureCredential` can be certain to work locally.